warrior/WarriorCore/testcase_steps_execution.py

```diff
@@ -9,239 +10,349 @@
+    print_info("************ This is parallel execution... console logs for {0} will be logged "
+               "in {1} ************".format(console_name, console_logfile))
     Utils.config_Utils.debug_file(console_logfile)
-
     return console_logfile
 
-def execute_steps(step_list, data_repository, system_name, parallel, queue):
-    """
-        Take in a list of steps
-        iterate through each of them and decide if each should run (pre-run check)
-        get status and report to term and log
-    """
-    default_error_action = data_repository['wt_def_on_error_action']
-    default_error_value = data_repository['wt_def_on_error_value']
 
-    goto_stepnum        = False
-    kw_resultfile_list  = []
-    step_status_list    = []
-    step_impact_list    = []
-    step_num            = 0
+class TestCaseStepsExecutionClass(object):
+    """ Step Execution Class """
 
-    if parallel is True:
-        system_console_log = get_system_console_log(data_repository['wt_filename'],
-                                                    data_repository['wt_logsdir'],
-                                                    '{0}_consoleLogs'.format(system_name))
-    while step_num < len(step_list):
-        step = step_list[step_num]
+    def __init__(self, step_list, data_repository, go_to_step_number, system_name, parallel, queue, skip_invoked=True):
+        """
+        Constructor for testcase_steps_execution_class.
+
+        :param step_list: List of steps
+        :param data_repository: Inital data_repository
+        :param go_to_step_number: go to step_number
+        :param system_name: System Name
+        :param parallel: parallel
+        :param queue: queue
+        :param skip_invoked: True by default
+        """
+        self.step_list = step_list
+        self.data_repository = data_repository
+        self.system_name = system_name
+        self.parallel = parallel
+        self.queue = queue
+        self.skip_invoked = skip_invoked
+        self.default_error_action = self.data_repository['wt_def_on_error_action']
+        self.default_error_value = self.data_repository['wt_def_on_error_value']
+        self.go_to_step_number = go_to_step_number
+        self.kw_resultfile_list = []
+        self.step_status_list = []
+        self.step_impact_list = []
+        self.current_step = False
+        self.current_step_number = False
+        self.next_step = False
+        self.run_current_step = False
+        self.current_triggered_action = False
+        self.step_status = None
+
+    def execute_step(self, current_step_number, go_to_step_number):
+        """
+        This function executes the determined step - step_num (integer index) from the step_list.
+        This function is called either from the while loop (normal execution) in function
+        execute_steps() or from a for loop (invoked execution)
+        """
+        self.current_step_number = current_step_number
```

Comment:
consider a list of three steps
[s1, s2, s3]

current implementation:
self.current_step  = s1
self.current_step_num = 0 i.e. the index of s1 in the list
self.next_step  =  1 (self.current_step_num + 1) .... the name is  misleading, it is not the next step but a meaningful number to the current step, in the code this is used so that we dont print list indexes like 0 but more meaningful step numbers in the console.

cange needed: add 1 to self.current_step_num and use it. Dont need self.next_step.

Reply:
Updated.